tests/integration_tests/device/cuda/navigation_validation.hpp

```diff
@@ -0,0 +1,304 @@
+template <typename bfield_t, typename detector_t,
+          typename intersection_record_t>
+inline auto run_navigation_validation(
+    vecmem::memory_resource *host_mr, vecmem::memory_resource *dev_mr,
+    detector_t *det,
+    const propagation::config<typename detector_t::scalar_type> &cfg,
+    bfield_t field_data,
+    const std::vector<std::vector<intersection_record_t>>
+        &host_intersection_traces)
+    -> vecmem::jagged_vector<
+        typename intersection_record_t::intersection_type> {
+
+    using intersection_t = typename intersection_record_t::intersection_type;
+
+    // Helper object for performing memory copies (to CUDA devices)
+    vecmem::cuda::copy cuda_cpy;
+
+    // Copy the detector to device and get its view
+    auto det_buffer = detray::get_buffer(*det, *dev_mr, cuda_cpy);
+    auto det_view = detray::get_data(det_buffer);
+
+    // Create vector buffer for intersection traces
+    std::vector<std::size_t> capacities;
+    for (const auto &trace : host_intersection_traces) {
+        capacities.push_back(trace.size());
+    }
+
+    // Create navigator candidates buffer
+    vecmem::cuda::managed_memory_resource mng_mr;
+    printf("Max candidates: %ld\n", det->n_max_candidates());
+    vecmem::copy copy;
+    auto navigation_buffer =
+        create_candidates_buffer(*det, host_intersection_traces.size(), mng_mr);
+    copy.setup(navigation_buffer);
+    /*auto navigation_buffer = detray::create_candidates_buffer(
+     *det, host_intersection_traces.size(), *dev_mr, host_mr);*/
+
+    std::cout << "buffer: " << navigation_buffer.size() << std::endl;
+
+    // Move intersection traces data to device
+    auto host_intersection_traces_data =
+        vecmem::get_data(host_intersection_traces, host_mr);
+    auto intersection_traces_buffer =
+        cuda_cpy.to(host_intersection_traces_data, *dev_mr, host_mr,
+                    vecmem::copy::type::host_to_device);
+    vecmem::data::jagged_vector_view<const intersection_record_t>
+        intersection_traces_view = vecmem::get_data(intersection_traces_buffer);
+
+    // Buffer for the intersection recorded by the navigator
+    vecmem::data::jagged_vector_buffer<intersection_t>
+        intersection_records_buffer(capacities, *dev_mr, host_mr,
+                                    vecmem::data::buffer_type::resizable);
+    cuda_cpy.setup(intersection_records_buffer);
+    auto intersection_records_view =
+        vecmem::get_data(intersection_records_buffer);
+
+    // Run the navigation validation test for GPU device
+    navigation_validation_device<bfield_t, detector_t, intersection_record_t>(
+        det_view, cfg, field_data, navigation_buffer, intersection_traces_view,
+        intersection_records_view);
+
+    // Get the result back to the host and pass it on to the checking
+    vecmem::jagged_vector<intersection_t> intersection_records(host_mr);
+    cuda_cpy(intersection_records_buffer, intersection_records);
+
+    return intersection_records;
```

Comment:
This is not a great setup by the way. Returning a vector that is created with a memory resource that's local to the function, is not too healthy. (Copy elision will return the actual object here. So are not even saved by vector copies not preserving the memory resource of the original object.)

Reply:
The managed memory resource was just there for testing, the function is being handed a host and device resource from the caller